android/src/reactNativeVersionPatch/nativeHierarchyManager/latest/com/swmansion/reanimated/layoutReanimation/ReanimatedNativeHierarchyManager.java

```diff
@@ -276,6 +286,12 @@ public synchronized void manageChildren(
 
     // we don't want layout animations in native-stack since it is currently buggy there
     if (viewGroupManager.getName().equals("RNSScreenStack")) {
+      if (indicesToRemove != null) {
+        for (int index : indicesToRemove) {
+          View child = viewGroupManager.getChildAt(viewGroup, index);
+          ((ReaLayoutAnimator) mReaLayoutAnimator).cancelAnimationsInSubviews(child);
+        }
+      }
```

Comment:
The fix for Android is to cancel all the animations in the transaction that removes the Screen.
This will cause all the exiting views to unregister their ancestors before being unmounted.